petsctools/__init__.py

```diff
@@ -30,3 +32,30 @@
         is_set_from_options,
         inserted_options,
     )
+else:
+
+    def __getattr__(name):
+        petsc4py_attrs = {
+            "get_blas_library",
+            "InvalidEnvironmentException",
+            "InvalidPetscVersionException",
+            "init",
+            "get_commandline_options",
+            "OptionsManager",
+            "petscobj2str",
+            "attach_options",
+            "has_options",
+            "get_options",
+            "set_from_options",
+            "is_set_from_options",
+            "inserted_options",
+        }
+        if name in petsc4py_attrs:
+            raise ImportError(
+                f"Cannot load '{name}' from module '{__name__}' because "
+                "petsc4py is not available"
```

Comment:
```suggestion
                f"Cannot load '{name}' from module '{__name__}' because "
                "petsc4py is not available.\n"
                "If this error appears during pip install then you may have"
                "forgotten to pass --no-build-isolation".
```